docs/ocf.md

```diff
@@ -101,7 +147,103 @@ the retrieved property data.
 Returns a promise which resolves successfully if there was no network error
 sending out the data.
 
-Sample Apps
+Server Samples
 -----------
 * [OCF Server sample](../samples/OcfServer.js)
-* [OCF Sensor sample](../samples/OcfSensorServer.js)
+* [OCF Sensor Server](../samples/OcfSensorServer.js)
+
+OCF Client
+----------
+```javascript
+// OCFClient is an EventEmitter
+interface Client {
+    Promise<Resource> findResources(ClientOptions options, optional FoundListener listener);
+    Promise<Resource> retrieve(string deviceId, object options);
+    Promise<Resource> update(Resource resource);
+    Promise<Platform> getPlatformInfo(string deviceId);
+    Promise<Device> getDeviceInfo(string deviceId);
+};
+
+dictionary ClientOptions {
+    string deviceId;
+    string resourceType;
+    string resourcePath;
+}
+
+interface Resource {
+
+    string resourcePath;      // Path for this resource
+    object properties;        // Application specific resource properties
+};
+
+callback FoundListener = void (ClientResource);
+
+```
+
+Client API Documentation
+------------------------
+### Client.findResources
+`Promise<ClientResource> findResources(ClientOptions options, optional FoundListener listener);`
+
+Find a remote resource matching `options` filter.
+
+The `options` paramter should contain a filter of resource options. Only
+resources matching theses options will be found.
+
+The `listener` paramter is an optional event listener callback. This
+callback will be called if a resource is found (`onfound` event).
+
+Returns a promise which resolves with a `ClientResource` object if a resource
+was found.
+
+### Client.retrieve
+`Promise<Resource> retrieve(string deviceId, object options);`
+
+Retrieve (GET) a remote resource.
+
+The `deviceId` parameter is the device ID of the resource you are retrieving.
+This ID must match a resource which has been found with `findResources()`.
+
+The `options` object properties contain flag information for this GET request
+e.g. `observable=true`.
+
+Returns a promise which resolves to a `ClientResource` containing the resource
+properties.
+
+### Client.update
+`Promise<Resource> update(Resource resource);`
+
+Update a remote resource properties.
+
+The `resource` parameter should contain a `deviceId` for the resource to
+update. The `properties` parameter will be sent to the resource and updated.
+
+Returns a Promise which resolves to a resource `Resource` containing the
```

Comment:
Here and twice in descriptions below you capitalize promise. Four times above you use 'promise'. Either is okay, just be consistent - (either you're referring to the conceptual promise or the specific class Promise). I'd probably just go with 'promise'.